cmd/command.go

```diff
@@ -20,36 +23,61 @@ func NewRootCommand() *cobra.Command {
 		},
 	}
 
-	cmd.PersistentFlags().StringVarP(
-		&app.configPath, "config-path", "c", "./merger.yaml",
-		"Path to the configuration file.")
-	cmd.PersistentFlags().IntVar(
-		&app.port, "listen-port", 8080,
-		"Listen port for the HTTP server.")
+	app.Bind(cmd)
 
 	cmd.AddCommand(NewVersionCommand())
 
 	return cmd
 }
 
 type App struct {
-	configPath string
-	port       int
+	viper *viper.Viper
 }
 
-func (app *App) run(cmd *cobra.Command, args []string) {
-	config, err := ReadConfig(app.configPath)
-	if err != nil {
-		log.WithField("error", err).Error("failed to load config")
-		return
-	}
+func (app *App) Bind(cmd *cobra.Command) {
+	app.viper = viper.New()
+	app.viper.SetEnvPrefix("MERGER")
+	app.viper.AutomaticEnv()
+
+	configPath := cmd.PersistentFlags().StringP(
+		"config-path", "c", "",
+		"Path to the configuration file.")
+	cobra.OnInitialize(func() {
+		if configPath != nil && *configPath != "" {
+			config, err := ReadConfig(*configPath)
+			if err != nil {
+				log.WithField("error", err).Error("failed to load config file")
```

Comment:
please add the configPath to the error message :)

Reply:
fixed